code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm

```diff
@@ -115,73 +115,73 @@
 	if(trigger_cold)
 		if(my_effect.trigger == TRIGGER_COLD && !my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_COLD && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_COLD && !secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 	else
 		if(my_effect.trigger == TRIGGER_COLD && my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_COLD && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_COLD && secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 
 	//HEAT ACTIVATION
 	if(trigger_hot)
 		if(my_effect.trigger == TRIGGER_HEAT && !my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_HEAT && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_HEAT && !secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 	else
 		if(my_effect.trigger == TRIGGER_HEAT && my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_HEAT && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_HEAT && secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 
 	//PHORON GAS ACTIVATION
 	if(trigger_phoron)
 		if(my_effect.trigger == TRIGGER_PHORON && !my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_PHORON && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_PHORON && !secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 	else
 		if(my_effect.trigger == TRIGGER_PHORON && my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_PHORON && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_PHORON && secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 
 	//OXYGEN GAS ACTIVATION
 	if(trigger_oxy)
 		if(my_effect.trigger == TRIGGER_OXY && !my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_OXY && !secondary_effect.activated)
-			secondary_effect.ToggleActivate(0)
+		if(secondary_effect.trigger == TRIGGER_OXY && !secondary_effect.activated)
+			secondary_effect.ToggleActivate()
 	else
 		if(my_effect.trigger == TRIGGER_OXY && my_effect.activated)
 			my_effect.ToggleActivate()
-		if(secondary_effect && secondary_effect.trigger == TRIGGER_OXY && !secondary_effect.activated)

```

Comment:
You get the gist.

Reply:
Because it's part of what's been making them impossible to trigger, if I'm correct. If the secondary effect AND the secondary effect trigger are trigger oxygen. The secondary effect won't be a trigger, it'll be something different, which is why I think the secondary effects haven't been working, because it's been relying on an if statement that's impossible to fulfill.

Reply:
No, what that means is
`if(secondary_effect)` -> which is basically a nullcheck
and
`if(secondary_effect.trigger == TRIGGER_OXY)`

Reply:
What you can do is change it to this:
`if(secondary_effect?.trigger == TRIGGER_OXY)`
which is the same thing.